### PR TITLE
introduce auto-generated dynamo client

### PR DESCRIPTION
**Pull request by rgarcia**
currently just used for one method: `GetWorkflowDefinition`

see https://github.com/Clever/wag/pull/175

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,11 +39,16 @@ swagger2markup-cli-1.3.1.jar:
 
 generate: wag-generate-deps swagger2markup-cli-1.3.1.jar
 	java -jar swagger2markup-cli-1.3.1.jar convert -c docs/config.properties -i swagger.yml  -d docs/
-	$(call wag-generate,./swagger.yml,$(PKG))
+	#$(call wag-generate,./swagger.yml,$(PKG))
+	# temporarily use a beta version of wag
+	bin/wag-beta -file swagger.yml -go-package $(PKG)/gen-go -js-path ./gen-js
```
Comment (nathanleiby) [inline Makefile]: nit: revert to non-beta once we've updated `wag`

File: store/dynamodb/dynamodb_store.go [hunk 2]
```diff
@@ -343,31 +346,39 @@ func (d DynamoDB) GetWorkflowDefinitionVersions(ctx context.Context, name string
...
+	// if len(res.Item) == 0 {
+	// 	return models.WorkflowDefinition{}, store.NewNotFound(name)
+	// }
+
+	// var wd models.WorkflowDefinition
+	// if err := DecodeWorkflowDefinition(res.Item, &wd); err != nil {
+	// 	return models.WorkflowDefinition{}, err
+	// }
+
+	// return wd, nil
```
Comment (nathanleiby) [inline store/dynamodb/dynamodb_store.go]: nit: remove commented code

File: store/tests/tests.go [hunk 3]
```diff
@@ -98,7 +98,8 @@ func GetWorkflowDefinition(s store.Store, t *testing.T) func(t *testing.T) {
 
 		_, err = s.GetWorkflowDefinition(ctx, "doesntexist", 1)
 		require.NotNil(t, err)
-		require.IsType(t, err, models.NotFound{})
+		// commenting this out for now, since new implementation does not use this type
+		//require.IsType(t, err, models.NotFound{})
```
Comment (nathanleiby) [inline store/tests/tests.go]: can we update test to verify it's the new error type?

File: Makefile (hunk 1 above)
Comment (rgarcia) [reply Makefile]: 426f02b

File: store/tests/tests.go (hunk 3 above)
Comment (rgarcia) [reply store/tests/tests.go]: 426f02b

File: store/dynamodb/dynamodb_store.go (hunk 2 above)
Comment (rgarcia) [reply store/dynamodb/dynamodb_store.go]: 27e616f

**Pull request merged**